### PR TITLE
fix(curriculum/python): improve shortest path algorithm project

### LINKED ISSUE #53395: Shortest Path Algorithm - Step 21 to 25 - Loop dictionaries not explained
### Describe the Issue

As a beginner in Python Steps 21 to 25 left me a bit confused and I had to look for external help to get some of them done.

Step 21 says:
`Create a for loop to iterate over your graph, and append each node to the unvisited list.`

One key information that is missing here is that when you loop through a dictionary the items (as in the x in _for x in list_) are the keys of the dict, not the whole key-value pair.

Until i got stumped in step 24 I thought the unvisited list contained the same information as the my_graph dictionary.

I am suggesting step 21 should be expanded a bit.

### Affected Page

https://www.freecodecamp.org/learn/scientific-computing-with-python/learn-algorithm-design-by-building-a-shortest-path-algorithm/step-21

### Your code

```
my_graph = {
    'A': [('B', 3), ('D', 1)],
    'B': [('A', 3), ('C', 4)],
    'C': [('B', 4), ('D', 7)],
    'D': [('A', 1), ('C', 7)]
}

def shortest_path(graph, start):
    unvisited = []
    for node in graph:
        unvisited.append(node)

```

### Expected behavior

An explanation of how looping dictionaries works.

### Screenshots

_No response_

### System

- Device: Laptop
- OS: Windows 11
- Browser: Firefox
- Version: 121.0.1 (64-bit)

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Dario-DC**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #53395

### CODE REVIEW

**Review by majestic-owl448 — CHANGES_REQUESTED** [review]
The title confuses me. What is algorithm design? I have not learn anything about that in this. I have learned about dictionaries, tuples and lists tho. Can't we change the name?

**Review by fhsinchy — APPROVED** [review]
I didn't find any issues or inconsistencies 👍

File: curriculum/challenges/english/07-scientific-computing-with-python/learn-algorithm-design-by-building-a-shortest-path-algorithm/6578b13757611e2825beb8a5.md [hunk 1]
```diff
@@ -9,26 +9,24 @@ dashedName: step-3
 
 Keys must be unique within a dictionary and they can be only immutable data types. This means you cannot use a list or another dictionary as keys.
 
-Add another key `age` to your dictionary and give it the integer number `2` as value.
+Add another key `'age'` to your dictionary and give it the integer number `2` as value.
 
 # --hints--
 
-You should add a new key-value pair to your `copper` dictionary.
+You should have an `'age'` key with the value `2` just after `'species': 'guinea pig'`, inside your `copper` dictionary. Don't forget the comma.
```
[the remark this answers is not in the document]
Comment (Dario-DC) [reply curriculum/challenges/english/07-scientific-computing-with-python/learn-algorithm-design-by-building-a-shortest-path-algorithm/6578b13757611e2825beb8a5.md:16]: The original name for this was "Learn Data Structures..." but the merge sort algorithm project has that name. 
I'm okay with changing the name, but not sure about which name we should use.

**Comment by Dario-DC** [comment]
> The title confuses me. What is algorithm design? I have not learn anything about that in this. I have learned about dictionaries, tuples and lists tho. Can't we change the name?

The original name for this was "Learn Data Structures..." but the merge sort algorithm project has that name. 
I'm okay with changing the name, but not sure about which name we should use.

**Comment by Dario-DC** [comment]
> The title confuses me. What is algorithm design? I have not learn anything about that in this. I have learned about dictionaries, tuples and lists tho. Can't we change the name?

Can we please get this merged and then deal with the new title in a separate PR? It's becoming a pain.

**Review by majestic-owl448 — APPROVED** [review]

**Comment by moT01** [comment]
Is this the way we typically do this? Don't we just make changes to English and then run an i18n sync after it gets merged?

**Comment by Dario-DC** [comment]
> Is this the way we typically do this? Don't we just make changes to English and then run an i18n sync after it gets merged?

@moT01 To be honest, I'm following what Naomi told me to do (we already did it a couple of times) but I don't understand the point of doing it. If you have an answer, you're very welcome.

I can get rid of the changes to the i18n folders, but then test will fail for Italian and Portuguese. Can we merge the PR then?

**Comment by moT01** [comment]
No, if Naomi said to do it like this - we should do it like this. I wasn't sure, that's why I asked.

**Review by majestic-owl448 — APPROVED** [review]

**Pull request merged**